test/e2e/framework/framework.go

```diff
@@ -0,0 +1,458 @@
+package framework
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/tools/remotecommand"
+
+	"github.com/submariner-io/admiral/pkg/federate/kubefed"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientset "k8s.io/client-go/kubernetes"
+
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// Polling interval while trying to create objects
+	PollInterval = 100 * time.Millisecond
+)
+
+type ClusterIndex int
+
+const (
+	ClusterA ClusterIndex = iota
+	ClusterB
+	ClusterC
+)
+
+const (
+	SubmarinerEngine = "submariner-engine"
+	GatewayLabel     = "submariner.io/gateway"
+)
+
+type PatchFunc func(pt types.PatchType, payload []byte) error
+
+type PatchStringValue struct {
+	Op    string `json:"op"`
+	Path  string `json:"path"`
+	Value string `json:"value"`
+}
+
```

Comment:
I know this file was copied from submariner but there's some types and code that isn't used/needed for lighthouse - I think it would be good to trim it down.

Reply:
This will be used in future tests, I have removed it from this patch.